src/guiguts/html_tools.py

```diff
@@ -1319,22 +1321,236 @@ def run(self) -> None:
 
         self.dialog.display_entries()
 
-    def get_title(self) -> str:
-        """Get sanitized book title from `<title>` element."""
-        title, nsub = re.subn(
-            r".+?<title>(.+?)</title>.+",
-            r"\1",
-            maintext().get("1.0", "20.0"),
-            flags=re.DOTALL,
+
+class EbookmakerCheckerAPIDialog(EbookmakerCheckerDialog):
+    """Dialog to show ebookmaker results using API."""
+
+    manual_page = "HTML_Menu#Ebookmaker"
+
+    def __init__(self, **kwargs: Any) -> None:
+        """Initialize ebookmaker dialog."""
+        super().__init__(
+            api=True,
+            **kwargs,
         )
-        if nsub == 0 or not title:
-            title = "No title"
-        title = re.sub(r"\s+", " ", title)
-        title = re.sub(r"\| Project Gutenberg", "", title)
-        title = re.sub(r"^\s+|\s+$", "", title)
-        title = DiacriticRemover.remove_diacritics(title)
-        title = re.sub(r"[^A-za-z0-9 ]+", "_", title)
-        return title
+
+
+class EbookmakerCheckerAPI:
+    """Ebookmaker checker using API."""
+
+    def __init__(self) -> None:
+        """Initialize ebookmaker checker."""
+
+        self.dialog = EbookmakerCheckerAPIDialog.show_dialog(
+            rerun_command=self.run,
+            sort_key_alpha=ebookmaker_sort_key_type,
+            show_suspects_only=True,
+        )
+
+    def run(self) -> None:
+        """Run ebookmaker using API."""
+        self.dialog.reset()
+
+        def report_exception(message: str, exc: Exception | str) -> None:
+            """Report exception to user and suggest manual conversion.
+            Also call `display_entries to clear "busy" message.
+
+            Args:
+                message: Initial part of message to user.
+                exc: Exception that was thrown, or a string describing the problem
+            """
+            logger.error(
+                f"{message}\nConvert manually online at ebookmaker.pglaf.org\nError details:\n{exc}"
+            )
+            self.dialog.display_entries()
+            self.dialog.lift()
+
+        timeout = 300
+
+        # Create list of formats to build
+        build = []
+        if preferences.get(PrefKey.EBOOKMAKER_EPUB2):
+            build.append("epub")
+        if preferences.get(PrefKey.EBOOKMAKER_EPUB3):
+            build.append("epub3")
+        if preferences.get(PrefKey.EBOOKMAKER_KINDLE):
+            build.append("kindle")
+        if preferences.get(PrefKey.EBOOKMAKER_KF8):
+            build.append("kf8")
```

Comment:
Might want to check that `build` isn't empty here if none of the prefs were set. I think the API will gladly let you ask for no `--make` inputs and pass that into ebookmaker which will probably succeed with an error message. Probably better to check that here first.

Reply:
In fact, in GG2, the Epub2 checkbox is turned on and disabled, so is always built. I'll initialize the array to reflect that.